webpack/assets/javascripts/react_app/components/Layout/index.js

```diff
@@ -12,27 +23,84 @@ import {
   selectIsLoading,
   selectIsCollapsed,
 } from './LayoutSelectors';
+import {
+  createInitialTaxonomy,
+  combineMenuItems,
+  getActiveMenuItem,
+} from './LayoutHelper';
+import { isLayoutCollapsed } from './LayoutSessionStorage';
 
 import Layout from './Layout';
 
-// map state to props
-const mapStateToProps = state => ({
-  items: patternflyMenuItemsSelector(state),
-  isLoading: selectIsLoading(state),
-  isCollapsed: selectIsCollapsed(state),
-  activeMenu: selectActiveMenu(state),
-  currentOrganization: selectCurrentOrganization(state),
-  currentLocation: selectCurrentLocation(state),
-});
+const ConnectedLayout = ({ children, data }) => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(
+      initializeLayout({
+        items: combineMenuItems(data),
+        activeMenu: getActiveMenuItem(data.menu).title,
+        isCollapsed: isLayoutCollapsed(),
+        organization:
+          createInitialTaxonomy(
+            data.orgs.current_org,
+            data.orgs.available_organizations
+          ) || ANY_ORGANIZATION_TAXONOMY,
+        location:
+          createInitialTaxonomy(
+            data.locations.current_location,
```

Comment:
I'm getting an error here (and component isn't rendered) if `data.orgs` or `data.locations` are null:
`Uncaught TypeError: Cannot read property 'current_location' of null`

Reply:
Nice catch, was that because your locations are disabled? If so it should work now...

Reply:
I have a user with no locations, no orgs, but it works now